packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/gollm/GoLLMController.java

```diff
@@ -330,77 +331,66 @@ public ResponseEntity<TaskResponse> createConfigFromDatasetTask(
-				log.warn(String.format("Dataset %s has no source files to send", datasetId));
-				throw new ResponseStatusException(HttpStatus.NOT_FOUND, messages.get("dataset.files.not-found"));
-			}
+		// Grab the dataset
+		final List<String> dataArray = new ArrayList<>();
 
-			for (final String filename : dataset.get().getFileNames()) {
-				try {
-					final Optional<String> datasetText = datasetService.fetchFileAsString(datasetId, filename);
-					if (datasetText.isPresent()) {
-						// ensure unescaped newlines are escaped
-						datasets.add(datasetText.get().replaceAll("(?<!\\\\)\\n", Matcher.quoteReplacement("\\\\n")));
-					}
-				} catch (final Exception e) {
-					log.warn("Unable to fetch dataset files", e);
-					throw new ResponseStatusException(HttpStatus.NOT_FOUND, messages.get("dataset.files.not-found"));
-				}
-			}
+		final Optional<Dataset> dataset = datasetService.getAsset(datasetId, permission);
+		if (dataset.isEmpty()) {
+			log.warn(String.format("Dataset %s not found", datasetId));
+			throw new ResponseStatusException(HttpStatus.NOT_FOUND, messages.get("dataset.not-found"));
 		}
 
-		if (datasets.isEmpty()) {
-			log.warn("No datasets found");
-			throw new ResponseStatusException(HttpStatus.NOT_FOUND, messages.get("dataset.not-found"));
+		// make sure there is text in the document
+		if (dataset.get().getFileNames() == null || dataset.get().getFileNames().isEmpty()) {
+			log.warn(String.format("Dataset %s has no source files to send", datasetId));
+			throw new ResponseStatusException(HttpStatus.NOT_FOUND, messages.get("dataset.files.not-found"));
 		}
 
-		// Grab the model
-		final Optional<Model> model = modelService.getAsset(modelId, permission);
-		if (model.isEmpty()) {
-			log.warn(String.format("Model %s not found", modelId));
-			throw new ResponseStatusException(HttpStatus.NOT_FOUND, messages.get("model.not-found"));
+		for (final String filename : dataset.get().getFileNames()) {
+			try {
+				final Optional<String> datasetText = datasetService.fetchFileAsString(datasetId, filename);
+				if (datasetText.isPresent()) {
+					final List<String> rows = Arrays.asList(datasetText.get().split("\\r?\\n"));
+					dataArray.addAll(rows);
+				}
+			} catch (final Exception e) {
+				log.warn("Unable to fetch dataset files", e);
+				throw new ResponseStatusException(HttpStatus.NOT_FOUND, messages.get("dataset.files.not-found"));
+			}
 		}
 
-		final ConfigureFromDatasetResponseHandler.Input input = new ConfigureFromDatasetResponseHandler.Input();
-		input.setDatasets(datasets);
+		final ConfigureModelFromDatasetResponseHandler.Input input = new ConfigureModelFromDatasetResponseHandler.Input();
+		input.setDataset(dataArray);
 		// stripping the metadata from the model before its sent since it can cause
 		// gollm to fail with massive inputs
 		model.get().setMetadata(null);
 		input.setAmr(model.get().serializeWithoutTerariumFields());
```

Comment:
Do we need to use `serializeWithoutTerariumFieldsKeepId` here to preserve the models id?

Reply:
nope, I now populate with the nil uuid if not found. We then postprocess and add the model id back in. caching can remain